### PR TITLE
Document Go actions.

### PR DESCRIPTION
**Pull request by rabbah**
Documents that you can run Go actions in OpenWhisk - which has been true for a while but not formally listed in our docs. This follows the support in place for executing any binary compatible executable.

### CODE REVIEW

File: docs/actions.md [hunk 1]
```diff
@@ -2,7 +2,12 @@
 # Creating and invoking OpenWhisk actions
 
 
-Actions are stateless code snippets that run on the OpenWhisk platform. An action can be written as a JavaScript, Swift, or Python function, a Java method, or a custom executable program packaged in a Docker container. For example, an action can be used to detect the faces in an image, respond to a database change, aggregate a set of API calls, or post a Tweet.
+Actions are stateless code snippets that run on the OpenWhisk platform.
+An action can be written as a JavaScript, Swift, Python or PHP function, a Java method,
+any binary-compatible executable including Go programs,
+or, a custom executable packaged in a Docker container.
```
Comment (dubeejw) [inline docs/actions.md]: `or,` to `or`.

File: docs/actions.md [hunk 2]
```diff
@@ -1107,6 +1114,70 @@ For the instructions that follow, assume that the Docker user ID is `janesmith`
   wsk action create example exec.zip --docker openwhisk/dockerskeleton
   ```
 
+## Creating Go actions
+
+The `--native` option allows for packaging of any executable as an action. This works for Go lang as an example.
```
Comment (dubeejw) [inline docs/actions.md]: `golang` as one word.

File: docs/actions.md [hunk 3]
```diff
@@ -1107,6 +1114,70 @@ For the instructions that follow, assume that the Docker user ID is `janesmith`
...
+    json.Unmarshal([]byte(arg), &obj)
+
+    // log to stdout 
+    fmt.Print("hello Go action")
+
+    name, ok := obj["name"].(string)
+    if !ok { name = "Stranger" }
+
+    // last line of stdout is the stringified result JSON object
+    // msg := map[string]string{"msg": ("Hello, " + name + "!")}
```
Comment (dubeejw) [inline docs/actions.md]: Uncomment this line?

File: docs/actions.md [hunk 4]
```diff
@@ -1107,6 +1114,70 @@ For the instructions that follow, assume that the Docker user ID is `janesmith`
...
+
+The action may be run as any other action.
+```bash
+ wsk action invoke helloGo -r -p name gopher 
+{
+    "msg": "Hello, gopher!"
+}
+```
+
+And logs are retrieved in a similar way as well.
```
Comment (dubeejw) [inline docs/actions.md]: Can drop `And` from this sentence.

File: docs/actions.md [hunk 5]
```diff
@@ -2,7 +2,12 @@
 # Creating and invoking OpenWhisk actions
 
 
-Actions are stateless code snippets that run on the OpenWhisk platform. An action can be written as a JavaScript, Swift, or Python function, a Java method, or a custom executable program packaged in a Docker container. For example, an action can be used to detect the faces in an image, respond to a database change, aggregate a set of API calls, or post a Tweet.
+Actions are stateless code snippets that run on the OpenWhisk platform.
+An action can be written as a JavaScript, Swift, Python or PHP function, a Java method,
+any binary-compatible executable including Go programs,
+or, a custom executable packaged in a Docker container.
+For example, an action can be used to detect the faces in an image, respond to a database change,
```
Comment (dubeejw) [inline docs/actions.md]: This sentence seems out of place. Maybe make it relate more to the above.

**Comment by dubeejw** [comment]
@jpkessle, can you take a look at the doc?

**Comment by rabbah** [comment]
Thanks - addressed.

File: docs/actions.md [hunk 6]
```diff
@@ -2,9 +2,16 @@
...
+Actions are stateless code snippets that run on the OpenWhisk platform.
+For example, an action can be used to detect the faces in an image, respond to a database change,
+aggregate a set of API calls, or post a Tweet.
+An action can be written as a JavaScript, Swift, Python or PHP function, a Java method,
+any binary-compatible executable including Go programs and custom executables packaged as Docker containers.
 
-Actions can be explicitly invoked, or run in response to an event. In either case, each run of an action results in an activation record that is identified by a unique activation ID. The input to an action and the result of an action are a dictionary of key-value pairs, where the key is a string and the value a valid JSON value. Actions can also be composed of calls to other actions or a defined sequence of actions.
+Actions can be explicitly invoked, or run in response to an event.
+In either case, each run of an action results in an activation record that is identified by a unique activation ID.
+The input to an action and the result of an action are a dictionary of key-value pairs, where the key is a string and the value a valid JSON value.
```
Comment (dubeejw) [inline docs/actions.md]: Regarding this line: `The input to an action and the result of an action are a dictionary of key-value pairs, where the key is a string and the value a valid JSON value.` Isn't all action input and output JSON? Maybe I'm just confused by it saying only the value is JSON.

Comment (rabbah) [reply docs/actions.md]: Note: This isn't new text - i just reformatted the text that was there. No semantic change.

JSON is a representation of a dictionary native to JS. In Swift and Python we use a dictionary type that is native to those languages.

**Pull request merged**